src/model/Space.cpp

```diff
@@ -2727,6 +2729,30 @@ namespace model {
       return result;
     }
 
+    double Space_Impl::exposedPerimeter(const Polygon3d& buildingPerimeter) const {
```

Comment:
Where do you get buildingPerimeter from?  Maybe it should get perimeter from the current story?

Reply:
Depending on your definitions buildingPerimeter could be multiple disjoint polygons.